### ISSUE TITLE
Bases just collapse for no reason.

### ISSUE THREAD

**Issue by GuguKatze** [kasami/kasamibot#6]
See the following 2 rooms for example:

Shard 2 / E36S1
Shard 2 / E35S8

It seems that a BaseHauler is missing after ExpansionWorkers quit their job and it goes downwards from there rapidly. The second room even has a completed storage.

Even by manually calling orderPioneer and order BaseHauler I don't seem to be able to restore "normality" ... :-( No containerMiners nor baseHaulers are queued by the AI ...

**Comment by kasami** [comment]
Yeah, the transition between the early phase with a lot of pioneers and the normal state of containerminers and haulers is buggy.

I've improved this for the next release.

The creeps should not just stand there as they do in E36S1 however, are you getting some errors in the console?

**Comment by GuguKatze** [comment]
Both rooms were dead creep wise (check the "replay") ... all the creeps there have been manually spawned or requested (as in the case of ExpansionWorkers). I played around with manually spawning Pioneers and BaseHaulers after my initial post.

I have  tracked the problem down ... it seems that the two bases are not returning "true" for 

if (!room.isExpansion()){}

in the managers.Mining.js script.

**Comment by kasami** [comment]
I've never played around with manually ordering units, the rooms should take care of that themselves.

You can fix the problem by removing the ConstructionSite for the tower. There's a bug where they keep trying to build that building, even now that the room is at RLC2 and not able to build it. The room should then start making new Pioneers to finish the construction-sites that has begun and upgrade the controller back to RCL3.

Addon:
And clear the order queue. (Game.rooms.E36S1.memory.orders = undefined)
It will do that over time, but not at once.

**Comment by GuguKatze** [comment]
Yeah, well ... the rooms were both totally void of creeps ... so I had to play around with calling orderPioneers and stuff manually. :-)

I suppose the Extension construction sites which can't yet be built (as of RCL2) are a problem as well.
